### PR TITLE
Resolve core Xcontent and CircuitBreaker refactor

### PR DESCRIPTION
**Pull request by lezzago**
*Issue #, if available:*
N/A

*Description of changes:*
Resolve core Xcontent refactor from https://github.com/opensearch-project/OpenSearch/pull/8826
Resolve core CircuitBreaker refactor from https://github.com/opensearch-project/OpenSearch/pull/9006

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/build.gradle [hunk 1]
```diff
@@ -115,6 +115,7 @@ dependencies {
     api project(":alerting-core")
     implementation "com.github.seancfoley:ipaddress:5.3.3"
 
+    testImplementation "org.antlr:antlr4-runtime:${versions.antlr4}"
```
Comment (eirsep) [inline alerting/build.gradle:118]: why is this required?

**Review by eirsep — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

File: alerting/build.gradle (hunk 1 above)
Comment (lezzago) [reply alerting/build.gradle:118]: This causes the integration tests to fail since its missing this dependency. This is because it is used by the Painless scripts and its caused by the BucketSelectorExtAggregator. I tried having this change in common-utils, but it wouldn't help.
